minid/minid.py

```diff
@@ -432,42 +425,27 @@ def get_or_register_rfm(self, rfm_record, test):
             "sha256": "6e3fbc3cc8c58edd0d99cd4925d18cdbd7ffbfa1a7fb201c06",
             "filename": "foo.txt"
           }
-
         """
-        log.debug('Checking entity {}'.format(rfm_record['filename']))
-        searchable_checksums = [ck_sum
-                                for alg_name, ck_sum in rfm_record.items()
-                                if alg_name in SUPPORTED_CHECKSUMS]
-        # Attempt to find any matching identifier for all hashes within
-        # the record.
-        # Break on the first checksum that returns results, and return the most
-        # recent hashes for that record.
-        entity = None
-        for checksum in searchable_checksums:
-            exst = self.identifiers_client.get_identifier_by_checksum(checksum)
-            minids = exst.data.get('identifiers', [])
-            entity = self.get_most_recent_active_entity(minids)
-            if entity:
-                break
-        if not entity:
-            checksums = [{'function': f, 'value': rfm_record.get(f)}
-                         for f in SUPPORTED_CHECKSUMS
-                         if f in rfm_record.keys()]
-            locations = (rfm_record['url']
-                         if isinstance(rfm_record['url'], list)
-                         else [rfm_record['url']]
-                         )
-            entity = self.register(checksums, test=test, locations=locations,
-                                   title=rfm_record['filename'])
+        checksums = [{'function': f, 'value': rfm_record.get(f)}
+                     for f in SUPPORTED_CHECKSUMS
+                     if f in rfm_record.keys()]
+        locations = (rfm_record['url']
+                     if isinstance(rfm_record['url'], list)
+                     else [rfm_record['url']]
+                     )
+        is_minid = self.is_minid(rfm_record['url'])
+        matches_namespace = self.is_test(rfm_record['url']) is test
+        if update_if_exists and is_minid and matches_namespace:
+            m_resp = self.update(rfm_record['url'], checksums=checksums,
+                               locations=locations).data
```

Comment:
Hmm, this isn't going to work due to minid not allowing checksums to change after initial creation. A slight update to this will be to check the minid first and continue with re-registration if the checksums do not match. If the checksum do match, it will update the existing record with any new locations.